generators/go/internal/generator/snippet.go

```diff
@@ -865,6 +865,20 @@ func (e *exampleContainerVisitor) VisitMap(mapContainer *ir.ExampleMapContainer)
 	return nil
 }
 
+func (e *exampleContainerVisitor) VisitNullable(nullable *ir.ExampleNullableContainer) error {
+	if nullable == nil || nullable.Nullable == nil {
+		return nil
+	}
+	e.value = &ast.Optional{
+		Expr: exampleTypeReferenceShapeToGoType(
+			nullable.Nullable.Shape,
+			e.types,
+			e.baseImportPath,
+		),
+	}
+	return nil
+}
```

Comment:
Do we need to prevent this from becoming a doubly wrapped optional?

Reply:
Yes indeed! In fact I neglected to address snippet generation entirely -- I've pushed an update for that. (Force pushed, so the diff to snippet.json is clear.)